src/mds/Server.cc

```diff
@@ -3506,10 +3506,10 @@ CInode* Server::prepare_new_inode(MDRequestRef& mdr, CDir *dir, inodeno_t useino
       _inode->mode |= S_ISGID;
     }
   } else {
-    _inode->gid = mdr->client_request->get_caller_gid();
+    _inode->gid = mdr->client_request->get_owner_gid();
   }
 
-  _inode->uid = mdr->client_request->get_caller_uid();
+  _inode->uid = mdr->client_request->get_owner_uid();
```

Comment:
Let's do a sanity check here to make sure the `owner_{uid/gid}` are correctly set. Which shouldn't be `-1`.

Reply:
Fixed, thanks!